src/functions/avl-datafeed-validator/index.ts

```diff
@@ -0,0 +1,175 @@
+import { createServerErrorResponse, createValidationErrorResponse } from "@bods-integrated-data/shared/api";
+import { getAvlSubscriptionErrorData } from "@bods-integrated-data/shared/avl/utils";
+import { getMetricStatistics } from "@bods-integrated-data/shared/cloudwatch";
+import { logger } from "@bods-integrated-data/shared/logger";
+import { AvlValidationError } from "@bods-integrated-data/shared/schema/avl-validation-error.schema";
+import { createStringLengthValidation } from "@bods-integrated-data/shared/validation";
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+import { ZodError, z } from "zod";
+import { getErrorDetail } from "./errorDetails";
+
+type AvlValidationErrorWithDetails = AvlValidationError & {
+    details: string;
+    critical: boolean;
+};
+
+type CategorisedErrors = {
+    criticalErrors: AvlValidationErrorWithDetails[];
+    nonCriticalErrors: AvlValidationErrorWithDetails[];
+};
+
+const criticalErrorFields = [
+    "Bearing",
+    "DestinationRef",
+    "LineRef",
+    "Monitored-VehicleJourney",
+    "OperatorRef",
+    "OriginRef",
+    "ProducerRef",
+    "RecordedAtTime",
+    "ResponseTimestamp",
+    "ValidUntilTime",
+    "VehicleJourneyRef",
+    "VehicleLocation",
+    "VehicleMonitoringDelivery",
+];
+
+const nonCriticalErrorFields = ["BlockRef", "DirectionRef", "OriginName", "PublishedLineName"];
```

Comment:
I feel like there is a simpler way to process the errors without declaring these extra types and mappings (and with the `categoriseErrors` function), since all the details about the error and error level are already included in dynamo